docs/controllers/kubeconfig.md

```diff
@@ -11,13 +11,21 @@ stateDiagram-v2
     [*] --> IsCurrentPlatformSupported
     state IsCurrentPlatformSupported <<choice>>
     IsCurrentPlatformSupported --> NoOp: False
-    IsCurrentPlatformSupported --> GetOperatorServiceAccount: True
-    GetOperatorServiceAccount --> GetServiceAccountSecret
-    GetServiceAccountSecret --> GenerateKubeconfigFromSecret
-    GenerateKubeconfigFromSecret --> CreateKubeconfigSecret
-    CreateKubeconfigSecret --> [*]
+    IsCurrentPlatformSupported --> GetOperatorServiceAccountSecret: True
+    GetOperatorServiceAccountSecret --> IsServiceAccountSecretFound
+    IsServiceAccountSecretFound --> IsServiceAccountSecretTooOld: True
+    IsServiceAccountSecretTooOld --> GenerateKubeconfig: False
+    GenerateKubeconfig --> [*]
+    IsServiceAccountSecretFound --> Requeue: False
+    Requeue --> GetOperatorServiceAccountSecret
+    IsServiceAccountSecretTooOld --> DeleterviceAccountSecret: True
+    DeleterviceAccountSecret --> Requeue
     NoOp --> [*]
 ```
 
 If the current platform is not supported, the controller will not create any secret and allow "bring your own" scenarios. 
 In cases where the platform is supported, the controller will create the secret containing kubeconfig.
+
+The controller will the service account secret that was initially created by the CVO. The token in the secret can exprire and has to
+be rotated. The controller will periodically check if the secret is too old and if so, it will delete the secret and wait for 
```

Comment:
I would appreciate actual link to controller code

Reply:
I included links for all controllers at the beginning of the doc